packages/core/client.js

```diff
@@ -314,6 +320,13 @@ class Client {
         event._handledState.severityReason = { type: 'userCallbackSetSeverity' }
       }
 
+      if (event.unhandled !== event._handledState.unhandled) {
+        event._handledState.severityReason.unhandledOverridden = true
+        event._handledState.unhandled = event.unhandled
+      }
+
+      trackSession(event)
```

Comment:
If my other comments are ratified then you can just inline contents of `trackSession` here as the logic is only used once.